### PR TITLE
Added support for multiple device provisioning

### PR DESCRIPTION
**Pull request by adgroc**
To enable users to easily provision multiple devices at once, support was added to import devices from a file. In the Windows client, users can now specify a csv file to import. The csv must contain the device id, and it can optionally contain the primary and secondary access keys. For each device id in the file, a new device will be provisioned in the specified iot hub.

In the nodejs cli, an import command was added. Users can specify a path to a text file that contains device ids on each line. For each device id specified a new device will be provisioned in the the specified iot hub.

### CODE REVIEW

**Comment by tameraw** [comment]
Hi @adgroc - Thanks for the contribution. Nice feature to have.

We'll update you as soon as we're done with the review of the pull request.

-Tamer

File: tools/iothub-explorer/doc/provision_device.md [hunk 1]
```diff
@@ -28,6 +28,32 @@ Created device mydevice
 ```
 Copy the device connection string information for later use.
 
+## Provision multiple devices in your IoT hub
+
+The iothub-explorer tool also enables you to provision multiple new devices at once. To provision multiple new devices:
+
+1. Create a text file and place the device id of each device you want to provision on a new line within the text file.
+2. Get the connection string for your IoT Hub. See [Setup IoT Hub][setup-iothub] for more details.
+3. Run the following commands in a terminal/shell window from the tools/iothub-explorer folder in the repository using the IoT Hub connection string from the previous step and replacing "mydevice" with your desired device name.
```
Comment (damonbarry) [inline tools/iothub-explorer/doc/provision_device.md]: "...using the IoT Hub connection string and text file path from the previous steps."  The bit about "mydevice" doesn't apply here.

File: tools/iothub-explorer/iothub-explorer.js [hunk 2]
```diff
@@ -165,6 +165,28 @@ else if (command === 'create') {
     }
   });
 }
+else if (command === 'import'){
+  if(!arg1) inputError('Import file not specified');
+  console.log(colorsTmpl('\n{green}Check if import file exists{/green}'));
```
Comment (damonbarry) [inline tools/iothub-explorer/iothub-explorer.js]: We don't want extraneous output in the "happy path" case.  Remove the log lines about checking for the import file, leaving only the inputError(...) when the import file doesn't exist.

File: tools/iothub-explorer/iothub-explorer.js [hunk 3]
```diff
@@ -275,6 +297,22 @@ else {
   usage();
 }
 
+function createDevice(deviceId){
```
Comment (damonbarry) [inline tools/iothub-explorer/iothub-explorer.js]: Can you update the 'create' command to call into this also, so we don't add redundant code to the file?  I think your createDevice function will need to be able to take either a deviceId or a JSON object (with a deviceId property) as an argument, but with that little modification it should work nicely for both commands.

**Comment by damonbarry** [comment]
What is iothub-explorer/Code?  GitHub reports that it's a 1.5 MB binary.  Unless I'm misunderstanding something, you'll need to scrub that binary out of your commits before we can take this contribution.

**Comment by damonbarry** [comment]
This PR includes code changes (as opposed to just docs), so we need to take this into develop, not master.

**Comment by adgroc** [comment]
Closing this pull request. Will issue another pull request for develop branch.

**Pull request closed without merging**